src/polynomial/mod.rs

```diff
@@ -19,10 +19,17 @@ pub struct Query {
     pub rotation: Rotation,
 }
 
+/// There are two types of variables in Expression:
+/// polynomial query ([`Expression::Polynomial`]) and challenge ([`Expression::Challenge`])
+/// we identify them by two global consts. It is used to identify the type of variable
+pub(crate) const POLYNOMIAL_TYPE: usize = 0;
+pub(crate) const CHALLENGE_TYPE: usize = 1;
```

Comment:
Why it's const, not enum options?

Reply:
Both are okay. It is easier to convert between const and usize. If use enum, we cannot simply use triple (i32, usize, usize).

Reply:
The approach with the representation of invariants by numbers "comes" from C and it is desirable for it to stay there. 

As soon as I separate indexes into a type, I will convert it to enum too